src/test/java/org/example/homework4/AppTest4.java

```diff
@@ -0,0 +1,120 @@
+package org.example.homework4;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.testng.annotations.*;
+import org.testng.asserts.SoftAssert;
+import resources.DataProviderClass;
+import resources.SingletonDriverClass;
+
+import java.util.*;
+
+/*
+        1 - create 3 negative tests in registration form*/
+
+public class AppTest4 {
+
+    String loginPath = "//div[@class='header-links'] //a[contains(@href, 'login')]";
+    String registerButPath = "//div[contains(@class, 'login-page')] //input[contains(@class,'register')]";
+    String submitRegButPath = "//div[contains(@class, 'registration')]//input[@type='submit']";
+    String fieldRegEmailPath = "//div[contains(@class, 'registration')]//input[@name='Email']";
+
+    String sumValidErrPath = "//div[contains(@class, 'registration')]//li";
+
+    String fieldRegFNamePath = "//div[contains(@class, 'registration')]//input[@name='FirstName']";
+    String fieldRegLNamePath = "//div[contains(@class, 'registration')]//input[@name='LastName']";
+    String fieldRegPhonePath = "//div[contains(@class, 'registration')]//input[@name='Phone']";
+    String fieldRegPassPath = "//div[contains(@class, 'registration')]//input[@name='Password']";
+    String fieldRegConfirmPassPath = "//div[contains(@class, 'registration')]//input[starts-with(@name,'ConfirmPass')]";
+    String validErrEmailXpath = "//div[contains(@class, 'registration')]//span[@id='Email-error']";
+
+
+    List<String> resultErrorMsgEmail = Arrays.asList(
+            "Невірне повідомлення електронної пошти",
+            "Електронна пошта обов'язкова."
+    );
+
+    Set<String> expectedValidErrorMsg = new HashSet<>(Arrays.asList("Ім'я обов'язкове.", "Прізвище обов'язкове.",
+            "Електронна пошта обов'язкова.", "телефон обов'язковий", "Password is required.", "Пароль обов'язковий."));
+
+    private WebDriver driver;
+
+    @BeforeSuite()
+    public void setup() {
+        this.driver = SingletonDriverClass.getInstanceOfSingletonDriverClass().getDriver();
+    }
+
+    @BeforeTest()
+
+
+    @BeforeClass()
+    public void actions() {
+        driver.get("https://butlers.ua/ua/");
+        WebElement login = driver.findElement(By.xpath(loginPath));
+        login.click();
+        WebElement register = driver.findElement(By.xpath(registerButPath));
+        register.click();
+    }
+
+
+    @Test(priority = 2, dataProvider = "EmailFieldNegativeData", dataProviderClass = DataProviderClass.class)
+    public void fieldEmailValidNegativeTest(String incorrect) {
+        SoftAssert soft = new SoftAssert();
```

Comment:
No need to create this object in each test's method. Create it ones on class level and use variable here and in tests below.